fe/src/main/java/org/apache/doris/catalog/Resource.java

```diff
@@ -0,0 +1,100 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog;
+
+import org.apache.doris.analysis.CreateResourceStmt;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.common.proc.BaseProcResult;
+import org.apache.doris.persist.gson.GsonUtils;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+
+public abstract class Resource implements Writable {
+    public enum ResourceType {
+        SPARK;
+
+        public static ResourceType fromString(String resourceType) {
+            for (ResourceType type : ResourceType.values()) {
+                if (type.name().equalsIgnoreCase(resourceType)) {
+                    return type;
+                }
+            }
+            return null;
+        }
+    }
+
+    @SerializedName(value = "name")
+    protected String name;
+    @SerializedName(value = "type")
+    protected ResourceType type;
+
+    public Resource(String name, ResourceType type) {
+        this.name = name;
+        this.type = type;
+    }
+
+    public static Resource fromStmt(CreateResourceStmt stmt) throws DdlException {
+        Resource resource = null;
+        ResourceType type = stmt.getResourceType();
+        switch (type) {
+            case SPARK:
+                resource = new SparkResource(stmt.getResourceName());
+                break;
+            default:
+                throw new DdlException("Only support Spark resource.");
+        }
+
+        resource.setProperties(stmt.getProperties());
+        return resource;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public ResourceType getType() {
+        return type;
+    }
+
+    protected abstract void setProperties(Map<String, String> properties) throws DdlException;
+    protected abstract void getProcNodeData(BaseProcResult result);
```

Comment:
should give comments for these functions to help others know how to implement them